devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp

```diff
@@ -0,0 +1,518 @@
+        : m_targetFrame(targetFrame)
+        , m_rootFrame(rootFrame)
+        , m_frameTransformInterface(frameTransformInterface)
+        , m_matrixBuffer(yarp::sig::Matrix(4, 4)){};
+
+    wearable::sensor::SensorStatus getStatus()
+    {
+        if (m_frameTransformInterface->canTransform(m_targetFrame, m_rootFrame)) {
+            return SensorStatus::Ok;
+        }
+        else {
+            return SensorStatus::Error;
+        }
+    }
+
+    bool getTransform(wearable::Quaternion& quaternion, wearable::Vector3& position)
+    {
+
+        if (!m_frameTransformInterface->getTransform(m_targetFrame, m_rootFrame, m_matrixBuffer)) {
+            return false;
+        }
+
+        position[0] = m_matrixBuffer(0, 3);
+        position[1] = m_matrixBuffer(1, 3);
+        position[2] = m_matrixBuffer(2, 3);
+
+        for (size_t r = 0; r < 3; r++) {
+            for (size_t c = 0; c < 3; c++) {
+                m_rotationMatrixBuffer[r][c] = m_matrixBuffer(r, c);
+            }
+        }
+
+        quaternion = wearable::utils::rotationMatrixToQuaternion(m_rotationMatrixBuffer);
+        return true;
+    }
+
+private:
+    yarp::dev::IFrameTransform* m_frameTransformInterface = nullptr;
+    std::string m_rootFrame, m_targetFrame;
+    yarp::sig::Matrix m_matrixBuffer;
+    wearable::Matrix3 m_rotationMatrixBuffer;
+};
+
+class IFrameTransformToIWear::Impl
+{
+public:
+    bool firstRun = true;
+    mutable std::recursive_mutex mutex;
+
+    TimeStamp timestamp;
+    ParsedOptions options;
+
+    yarp::dev::IFrameTransform* frameTransformInterface = nullptr;
+
+    template <typename T>
+    using SensorsMap = std::map<std::string, SensorPtr<T>>;
+
+    // Sensors stored for exposing wearable::IWear
+    class PoseSensor;
+    SensorsMap<PoseSensor> poseSensorsMap;
+
+    template <typename SensorType>
+    bool isSensorAvailable(const std::string& name, const SensorsMap<SensorType>& map)
+    {
+        if (map.find(name) == map.end()) {
+            return false;
+        }
+        return true;
+    }
+};
+
+// ================================
+// WEARABLE SENSORS IMPLEMENTATIONS
+// ================================
+
+class IFrameTransformToIWear::Impl::PoseSensor : public wearable::sensor::IPoseSensor
+{
+public:
+    // ------------------------
+    // Constructor / Destructor
+    // ------------------------
+    PoseSensor(
+        IFrameTransformHandler* frameTransformHandler,
+        const wearable::sensor::SensorName aName,
+        const wearable::sensor::SensorStatus aStatus = wearable::sensor::SensorStatus::Unknown)
+        : IPoseSensor(aName, aStatus)
+        , m_frameTransformHandler(frameTransformHandler)
+    {}
+
+    ~PoseSensor() override = default;
+
+    // -------------------------
+    // IOrientationSensor interface
+    // -------------------------
+    bool getPose(Quaternion& orientation, Vector3& position) const override
+    {
```

Comment:
I see the point of having a mutex here, but this method belongs to a different class w.r.t. where the [impl mutex is defined](https://github.com/robotology/wearables/blob/add-iFrameTransformToIWear-device/devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp#L82). I am not sure how to proceed to add the mutex here, should we have a different mutex local to [`PoseSensor`](https://github.com/robotology/wearables/blob/add-iFrameTransformToIWear-device/devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp#L119) class, or should we pass the same mutex?
cc @diegoferigo

Reply:
probably, what you can do is that, pass the pointer to the `Impl`class to it. Add a new member variable to the class, like:
`    IFrameTransformToIWear::Impl* m_frameTransformImpl{nullptr};` and you can initialize in the constructor. In this way, you can have access to the same `mutex` object of the implementation.

Reply:
Historically we always used a single mutex to protect the access of all sensor implementations for a single device. The mutex is typically stored in its pImpl. This is convenient if the storage of the sensors is shared among all sensors types and a global protection is needed.

In this specific case, at first sight, it should not be a big problem because the data is gathered externally from the transform server (i.e. there is no one that calls a `setTransform` to the buffer that could interfere with the `getTransform`). However, multiple calls of `getTransform` could theoretically occurr, what would happen if both threads execute almost together? I fear that it could corrupt data, maybe protecting the `getTrasform` call with a mutex (that could be done in the `getPose` method) keeps us on the safe zone.

Reply:
mutex had been added to protect `getPose` method (and indeed the call to `iFrameTransform::getTransform()`) with https://github.com/robotology/wearables/pull/126/commits/faf2bbce350a99f7ecc78983839a932fc5d977be